src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -274,6 +274,18 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer  { thisTran
               case tpe => tpe
             }
           )
+        case Import(expr, selectors) =>
+          val exprTpe = expr.tpe
+          def notExists(name: Name): Boolean =
+            name != nme.WILDCARD && !exprTpe.member(name).exists
```

Comment:
I think you need to refine this so that you look for term names and for type names. 

I also agree that it should be an error.